### PR TITLE
Fix error on getting type from metadata when no type provided

### LINKED ISSUE #229: TypeError: Cannot read property '0' of undefined in ParamMetadata.ts:117
C:\wamp\vdrive\web\dev-jobz\parser\src\metadata\ParamMetadata.ts:117
this.targetType = args.explicitType ? args.explicitType : (Reflect as any).getMetadata("design:paramtypes", args.object, args.method)[args.index];
^
TypeError: Cannot read property '0' of undefined
at new ParamMetadata (C:\wamp\vdrive\web\dev-jobz\parser\src\metadata\ParamMetadata.ts:117:142)
at C:\wamp\vdrive\web\dev-jobz\parser\src\metadata-builder\MetadataBuilder.ts:101:31
at Array.map (native)
at MetadataBuilder.createParams (C:\wamp\vdrive\web\dev-jobz\parser\src\metadata-builder\MetadataBuilder.ts:101:14)
at C:\wamp\vdrive\web\dev-jobz\parser\src\metadata-builder\MetadataBuilder.ts:87:38
at Array.map (native)
at MetadataBuilder.createActions (C:\wamp\vdrive\web\dev-jobz\parser\src\metadata-builder\MetadataBuilder.ts:85:14)
at C:\wamp\vdrive\web\dev-jobz\parser\src\metadata-builder\MetadataBuilder.ts:72:39
at Array.map (native)
at MetadataBuilder.createControllers (C:\wamp\vdrive\web\dev-jobz\parser\src\metadata-builder\MetadataBuilder.ts:69:28)

Maybe it's a problem in my code, but how do I know where is the problem?

### PR DESCRIPTION
**Pull request by 19majkel94**
Fixes #229.

### CODE REVIEW

File: src/metadata/ParamMetadata.ts [hunk 1]
```diff
@@ -114,7 +114,14 @@ export class ParamMetadata {
         this.transform = args.transform;
         this.classTransform = args.classTransform;
         this.validate = args.validate;
-        this.targetType = args.explicitType ? args.explicitType : (Reflect as any).getMetadata("design:paramtypes", args.object, args.method)[args.index];
+        if (args.explicitType) {
+			this.targetType = args.explicitType
+		} else {
+			const metadata = (Reflect as any).getMetadata("design:paramtypes", args.object, args.method);
+			if (typeof metadata != "undefined") {
```
Comment (NoNameProvided) [inline src/metadata/ParamMetadata.ts]: Dont we have to set `this.targetType` to something if `metadata` is undefined? I am just wondering, cannot cause an other error later if we leave it `undefined`?

File: src/metadata/ParamMetadata.ts [hunk 2]
```diff
@@ -114,7 +114,14 @@ export class ParamMetadata {
         this.transform = args.transform;
         this.classTransform = args.classTransform;
         this.validate = args.validate;
-        this.targetType = args.explicitType ? args.explicitType : (Reflect as any).getMetadata("design:paramtypes", args.object, args.method)[args.index];
+        if (args.explicitType) {
+			this.targetType = args.explicitType
```
Comment (NoNameProvided) [inline src/metadata/ParamMetadata.ts]: This needs some ident fixing, and removing the empty line.

**Comment by NoNameProvided** [comment]
Ping @spidgorny to see if he can do the re-formatting and also answer the question.

File: src/metadata/ParamMetadata.ts (hunk 1 above)
Comment (19majkel94) [reply src/metadata/ParamMetadata.ts]: It shouldn't, we have a check `if (this.targetType)` a few lines below.

Comment (NoNameProvided) [reply src/metadata/ParamMetadata.ts]: Ok. Then it's ok for me. However we should fix the indentation issue after merging.

**Comment by 19majkel94** [comment]
Ok, so I'm merging and I will do reformatting/renaming on master in a minutes.

**Comment by 19majkel94** [comment]
`if (typeof ParamTypes !== "undefined")`
I don't know why just no `if (ParamTypes)` but let's leave it as it works and it's readable 😉

**Pull request merged**